### ISSUE TITLE
Proposition for conditional stage execution

### ISSUE THREAD

**Issue by ribab** [open-mpi/mtt#919]
Hello MTT community,

I have a requirement to use MTT to run the same INI file on two different systems that have two different resource managers (SLURM and PBSPro). An elegant solution for this would be to add an option that works for all MTT stages for optionally running the stage based on the presence and value of an environment variable or file. This way all INI files will continue to work as-is, and we just need to add a PBSPro launcher for supporting the new resource manager that will be optionally enabled.

The option **run_if** will run a shell command, and if the return code is 0 the section is ran, otherwise the section is skipped.

Here is an INI snippet to show how **run_if** option would be used

```
[LauncherDefaults:SLURM]
run_if = test $$RESOURCE_MGR == slurm
plugin = SLURM
parent = TestBuild:OtherSection
command = salloc
options = -w ${ENV:SETH_NODE_LIST}
          -p ${ENV:PARTION_TYPE}

[LauncherDefaults:PBSPro]
run_if = test $$RESOURCE_MGR == pbspro
plugin = PBSPro
parent = TestBuild:OtherSection
command = <pbspro allocate command>
options = <pbspro options>
```

I would like to know what you think about this proposal?

**Comment by ribab** [comment]
The implementation for this feature would be to add a check on this line

https://github.com/open-mpi/mtt/blob/1f5daed408aaaaa2395e82b5afcf749373b7c960/pylib/Tools/Executor/sequential.py#L279

to check this I will add a check for `keyvals['run_if']` and if it exists then run the shell command included in that option to determine whether the stage should be skipped. To skip the stage I just need to run `continue` in sequential.py

**Comment by hppritcha** [comment]
I like this!  Great add.  I'd be able to use this as well.

**Comment by beardeddog** [comment]
Ricky, so what is the stage status for a stage that fails the test?  skipped or something else?

**Comment by ribab** [comment]
The plugin is what sets the stage status, and since the plugin is being skipped, I was thinking no entry for that stage would be seen in the log and we would only see the skip information in the verbose output.

**Comment by beardeddog** [comment]
ok, yeah I see that in the test you ran, looks fine

**Comment by rohlem** [comment]
To me it looks like this was implemented by merging PR #921 , is that correct? Is this issue intentionally still open?

I would also like to point out that the added documentation in [`docs/pages/user_guide.md`](https://github.com/open-mpi/mtt/blob/master/docs/pages/user_guide.md) is not yet reflected on [the corresponding page on the Github-io-site](https://open-mpi.github.io/mtt/pages/user_guide.html) (which seems to me auto-generated from that source file).
Is updating that page currently a manual process, or is there supposed to be automation in place that missed the change when it happened?

**Comment by hppritcha** [comment]
@rohlem yes PR #921 addresses this issue.  Closing now.

As to the github io pages question I thought that was being automatically updated with changes.  I'll investigate.
